### PR TITLE
feature: #36 | Add the Error codes and Rest Exception

### LINKED ISSUE #36: Add the Error codes and Rest Exception

### PR DESCRIPTION
**Pull request by astha-singh-dev**
closes #36

### CODE REVIEW

File: src/main/java/dev/shivamnagpal/users/dtos/response/ResponseWrapper.java [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
 import lombok.Getter;
 import lombok.Setter;
 
+import java.util.List;
+
 @AllArgsConstructor
 @Getter
 @Setter
 public class ResponseWrapper<T> {
+    List<ErrorResponse> errors;
```
Comment (ShivamNagpal) [inline src/main/java/dev/shivamnagpal/users/dtos/response/ResponseWrapper.java]: Make this field private

File: src/main/java/dev/shivamnagpal/users/dtos/response/ResponseWrapper.java [hunk 2]
```diff
@@ -18,15 +23,19 @@ public class ResponseWrapper<T> {
...
     public static <T> ResponseWrapper<T> failure(T payload, String message) {
-        return new ResponseWrapper<>(false, payload, message);
+        return new ResponseWrapper<>(null, false, payload, message);
+    }
+
+    public static <T> ResponseWrapper<T> failure(T payload, String message, List<ErrorResponse> errors) {
+        return new ResponseWrapper<>(errors, false, payload, message);
     }
 
     public static <T> ResponseWrapper<T> failure(T payload) {
```
Comment (ShivamNagpal) [inline src/main/java/dev/shivamnagpal/users/dtos/response/ResponseWrapper.java]: Only 1 failure overload is needed for now. We don't need `payload` anymore for failure cases.

```public static <T> ResponseWrapper<T> failure(List<ErrorResponse> errorResponses) {}```

File: src/main/java/dev/shivamnagpal/users/exceptions/handlers/GlobalExceptionHandler.java [hunk 3]
```diff
@@ -19,18 +15,10 @@ private GlobalExceptionHandler() {
...
-                        MessageConstants.RESPONSE_EXCEPTION_PAYLOAD,
-                        Json.encodePrettily(responseExceptionPayload)
-                );
-                log.error(message);
-            }
+        if (throwable instanceof RestException restException) {
             ResponseWrapper<Object> failureResponseWrapper = ResponseWrapper
-                    .failure(responseExceptionPayload, responseException.getMessage());
-            httpServerResponse.setStatusCode(responseException.getStatus())
+                    .failure(null, restException.getMessage());
```
Comment (ShivamNagpal) [inline src/main/java/dev/shivamnagpal/users/exceptions/handlers/GlobalExceptionHandler.java]: Get the errorResponses from the RestException and set it in the ResponseWrapper failure.

**Review by ShivamNagpal — CHANGES_REQUESTED** [review]

File: src/main/java/dev/shivamnagpal/users/dtos/response/ResponseWrapper.java (hunk 1 above)
Comment (astha-singh-dev) [reply src/main/java/dev/shivamnagpal/users/dtos/response/ResponseWrapper.java]: fixed

File: src/main/java/dev/shivamnagpal/users/exceptions/handlers/GlobalExceptionHandler.java (hunk 3 above)
Comment (astha-singh-dev) [reply src/main/java/dev/shivamnagpal/users/exceptions/handlers/GlobalExceptionHandler.java]: fixed

File: src/main/java/dev/shivamnagpal/users/dtos/response/ResponseWrapper.java (hunk 2 above)
Comment (astha-singh-dev) [reply src/main/java/dev/shivamnagpal/users/dtos/response/ResponseWrapper.java]: add the above factory method, there is another story for response wrapper, will remove other failure method in that

**Review by ShivamNagpal — APPROVED** [review]

**Pull request merged**